### PR TITLE
Update config docs (uwsgi >> gunicorn)

### PR DESCRIPTION
**Pull request by jdavcs**
#12257 

(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by jmchilton — APPROVED** [review]

File: doc/source/admin/config.rst [hunk 1]
```diff
@@ -93,22 +93,23 @@ Configuration Basics
 ----------------------------
 
 - Edit ``config/galaxy.yml`` (copy it from ``config/galaxy.yml.sample`` if it does not exist) to make configuration
-  changes. This is a `uWSGI YAML configuration file`_ and should contain two sections, one named ``uwsgi`` for uWSGI and
-  one named ``galaxy`` for Galaxy.
+  changes. This is a `YAML configuration file`_ and should contain one or two sections, one named ``gravity`` for
+  Gunicorn server options (optional) and one named ``galaxy`` for Galaxy (required).
```
Comment (nsoranzo) [inline doc/source/admin/config.rst]:
```suggestion
  process management via `Gravity`_ (optional) and one named ``galaxy`` for Galaxy (required).
```

File: doc/source/admin/config.rst [hunk 2]
```diff
@@ -93,22 +93,23 @@ Configuration Basics
...
 
     - The default port for the Galaxy web server is ``8080``, and it only binds to localhost by default. To configure
```
Comment (nsoranzo) [inline doc/source/admin/config.rst]:
```suggestion
    - The `gravity` section contain several important options about how to
      launch Galaxy web workers and job handlers, which are documented in the
      :doc:`Scaling and Load Balancing <scaling>` documentation.
```

File: doc/source/admin/config.rst [hunk 3]
```diff
@@ -93,22 +93,23 @@ Configuration Basics
...
-      0.0.0.0:8080``).
-    - Some uWSGI options are required for uWSGI to run Galaxy properly and will be added to the ``uwsgi`` command line
-      by ``run.sh`` if not specified in ``galaxy.yml``.
-    - uWSGI has a `large number of options`_. The Galaxy documentation refers to some of them, but many additional
+      Gunicorn to listen on all available network addresses, set ``bind`` in the ``gravity`` section of ``galaxy.yml``
+      to ``0.0.0.0:<port>`` (e.g. ``bind: 0.0.0.0:8080``).
+    - Some Gunicorn options are required for Gunicorn to run Galaxy properly and will be added to the ``gunicorn``
+      command line by `Gravity` via the ``run.sh`` script if not specified in ``galaxy.yml``.
+    - Gunicorn has a `large number of options`_. The Galaxy documentation refers to some of them, but many additional
       advanced deployment scenarios are available.
```
Comment (nsoranzo) [inline doc/source/admin/config.rst]:
```suggestion
```

File: doc/source/admin/config.rst [hunk 4]
```diff
@@ -93,22 +93,23 @@ Configuration Basics
...
 
 - Run Galaxy with ``sh run.sh``
 - Use a web browser and go to the address you configured in ``galaxy.yml`` (defaults to http://localhost:8080/)
 
 .. _uWSGI YAML configuration file: https://uwsgi-docs.readthedocs.io/en/latest/Configuration.html
-.. _large number of options: https://uwsgi-docs.readthedocs.io/en/latest/Options.html
+.. _large number of options: https://docs.gunicorn.org/en/latest/settings.html
```
Comment (nsoranzo) [inline doc/source/admin/config.rst]:
```suggestion
```

File: doc/source/admin/config.rst [hunk 5]
```diff
@@ -93,22 +93,23 @@ Configuration Basics
...
+      to ``0.0.0.0:<port>`` (e.g. ``bind: 0.0.0.0:8080``).
+    - Some Gunicorn options are required for Gunicorn to run Galaxy properly and will be added to the ``gunicorn``
+      command line by `Gravity` via the ``run.sh`` script if not specified in ``galaxy.yml``.
+    - Gunicorn has a `large number of options`_. The Galaxy documentation refers to some of them, but many additional
       advanced deployment scenarios are available.
 
 - Run Galaxy with ``sh run.sh``
 - Use a web browser and go to the address you configured in ``galaxy.yml`` (defaults to http://localhost:8080/)
 
 .. _uWSGI YAML configuration file: https://uwsgi-docs.readthedocs.io/en/latest/Configuration.html
```
Comment (nsoranzo) [inline doc/source/admin/config.rst]:
```suggestion
```

File: doc/source/admin/config.rst (hunk 3 above)
Comment (jdavcs) [reply doc/source/admin/config.rst]: You mean we should provide these options and their documentation via Gravity?

Comment (jdavcs) [reply doc/source/admin/config.rst]: Oh I see, your previous edit. Never mind.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**